tests/console/network.pm

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Check for network daemon
+# Maintainer: Dominik Heidler <dheidler@suse.de>
+
+use base 'consoletest';
+use strict;
+use testapi;
+use utils;
+use version_utils;
+
+sub run {
+    select_console 'root-console';
+
+    assert_script_run 'ip a';
+
+    if (is_opensuse) {
+        # check for systemd-networkd
+        systemctl 'is-enabled systemd-networkd', expect_false => 1;
+        systemctl 'is-active systemd-networkd',  expect_false => 1;
+        assert_script_run 'networkctl status';
+    }
+
+    my $network_daemon = script_output 'readlink /etc/systemd/system/network.service | sed \'s#.*/\(.*\)\.service#\1#\'';
+
+    if (check_var('DESKTOP', 'textmode') && $network_daemon ne 'wicked') {
```

Comment:
What's about rewriting it as follows not to do same checks twice:
```perl
    if (check_var('DESKTOP', 'textmode')) {
        die "Expected wicked for DESKTOP=textmode but got '$network_daemon'" if $network_daemon ne 'wicked';       
    }
    else {
        die "Expected NetworkManager for DESKTOP!=textmode but got '$network_daemon'" if $network_daemon ne 'NetworkManager';
    }
```
Also, I thought for SLES we always get wicked, so maybe we need to inject more complex condition for expectations. NM is default for SLED though.

Reply:
Updated the patch as suggested.

Reply:
Fixed the code to expect nm for SLED and wicked for SLES.

Reply:
I'm just not sure where to schedule the test for SLED as SLED doesn't have `extra_tests`.
And how should we handle this with SLE15 where desktop will just be an addon?

@okurz WDYT?